doc/en/conf.py

```diff
@@ -433,29 +237,33 @@
     )
 ]
 
-# -- Options for towncrier_draft extension -----------------------------------
+# -- Options for towncrier_draft extension --------------------------------------------
+# https://sphinxcontrib-towncrier.readthedocs.io/en/latest/#how-to-use-this
 
 towncrier_draft_autoversion_mode = "draft"  # or: 'sphinx-version', 'sphinx-release'
 towncrier_draft_include_empty = True
 towncrier_draft_working_directory = PROJECT_ROOT_DIR
 towncrier_draft_config_path = "pyproject.toml"  # relative to cwd
 
-
 # -- Options for sphinx_issues extension -----------------------------------
+# https://github.com/sloria/sphinx-issues#installation-and-configuration
 
 issues_github_path = "pytest-dev/pytest"
 
+# -- Custom Read the Docs build configuration -----------------------------------------
+# (undocumented, relies on implementation details of Read the Docs' build)
 
-intersphinx_mapping = {
-    "pluggy": ("https://pluggy.readthedocs.io/en/stable", None),
-    "python": ("https://docs.python.org/3", None),
-    "numpy": ("https://numpy.org/doc/stable", None),
-    "pip": ("https://pip.pypa.io/en/stable", None),
-    "tox": ("https://tox.wiki/en/stable", None),
-    "virtualenv": ("https://virtualenv.pypa.io/en/stable", None),
-    "setuptools": ("https://setuptools.pypa.io/en/stable", None),
-    "packaging": ("https://packaging.python.org/en/latest", None),
-}
+IS_RELEASE_ON_RTD = (
+    os.getenv("READTHEDOCS", "False") == "True"
+    and os.environ["READTHEDOCS_VERSION_TYPE"] == "tag"
+)
+if IS_RELEASE_ON_RTD:
+    tags: set[str]
+    # pylint: disable-next=used-before-assignment
+    tags.add("is_release")  # noqa: F821
```

Comment:
Possibly unrelated to this PR, but I don't think this does anything other than make the docs build fail on tags for RtD?

Looking at the debug view in https://app.readthedocs.org/projects/pytest/builds/24926485/, I can see that `cat` of the `conf.py` file shows that the readthedocs.org specific logic is "magically" injected _after_ this conf.py file's contents. This means that the `tags` isn't a variable that is set at this point and this will unconditionally fail. 

Let me know if it would be acceptable to remove this entirely.

Reply:
@pradyunsg don't touch this bit. It's a variation of what you did in pip for hiding the draft changelog part but implemented on a different level. You were modifying the Towncrier template IIRC but this is used to let `.. only::` skip calling the `sphinxcontrib-towncrier` extension on tagged builds. This feature of Sphinx was already in the repo before I integrated my extension. I learned it from here. However, they used to call something like `sphinx-build -t is_release` and I came up with the idea of setting it automatically. I've been using it for a while, in several other repos before I got to integrating my extension into Pytest. Though, I may move this chunk inside `setup()` at some point and avoid this whole thing with relying on the magic of Sphinx that injects `tags` into `globals()` of `conf.py`. Though, FWIW, this is intended/documented use. It does not fail with `NameError` (despite what the linters assume).

Reply:
Oh wow.

https://github.com/sphinx-doc/sphinx/blob/v7.3.7/sphinx/config.py#L502